### PR TITLE
Bump Diplomat and use `DiplomatChar`

### PR DESCRIPTION
**Pull request by robertbastian**
~The panics in this code are not new, they were in the Diplomat glue before. However, we should still remove them.~

Removed all panics

#2520

### CODE REVIEW

File: components/collections/src/codepointinvlist/builder.rs [hunk 1]
```diff
@@ -221,8 +221,12 @@ impl CodePointInversionListBuilder {
     /// let check = builder.build();
     /// assert_eq!(check.iter_chars().next(), Some('B'));
     pub fn remove_char(&mut self, c: char) {
-        let to_remove = c as u32;
-        self.remove(to_remove, to_remove + 1);
+        self.remove_u32(c as u32)
+    }
+
+    /// See [`Self::remove_char`]
+    pub fn remove_u32(&mut self, c: u32) {
```
Comment (sffc) [inline components/collections/src/codepointinvlist/builder.rs:228]: Issue: these functions should be named `foo32` if they take a u32 argument and `foo_u32` if they return a u32. Previous discussion: https://github.com/unicode-org/icu4x/issues/2413#issuecomment-1227572950

**Review by sffc — CHANGES_REQUESTED** [review]

File: components/collections/src/codepointinvlist/builder.rs (hunk 1 above)
Comment (robertbastian) [reply components/collections/src/codepointinvlist/builder.rs:228]: There are existing `foo_u32` methods here. Do we want to break consistency?

**Review by Manishearth — APPROVED** [review]

File: components/collections/src/codepointinvlist/builder.rs (hunk 1 above)
Comment (sffc) [reply components/collections/src/codepointinvlist/builder.rs]: Yeah I think we should try to follow the style we agreed upon. I don't know why the functions like `add_u32` were not changed in #2460.

Comment (sffc) [reply components/collections/src/codepointinvlist/builder.rs]: https://github.com/unicode-org/icu4x/issues/4354

**Pull request merged**